cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperEraser.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * Deletes the regular CLusterIP service used by ZooKeeper clients
+     *
+     * @return  Completes when the service was successfully deleted
+     */
+    protected Future<Void> service() {
+        return serviceOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperServiceName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletes the headless service
+     *
+     * @return  Completes when the service was successfully deleted
+     */
+    protected Future<Void> headlessService() {
+        return serviceOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperHeadlessServiceName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletes the Secret with the node certificates used by the ZooKeeper nodes.
+     *
+     * @return      Completes when the Secret was successfully deleted
+     */
+    protected Future<Void> certificateSecret() {
+        return secretOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperSecretName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletes the ConfigMap with logging and metrics configuration.
+     *
+     * @return  Completes when the ConfigMap was successfully deleted
+     */
+    protected Future<Void> loggingAndMetricsConfigMap() {
+        return configMapOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperMetricsAndLogConfigMapName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletes the PodDisruptionBudgets on Kubernetes clusters which support v1 version of PDBs
+     *
+     * @return  Completes when the PDB was successfully deleted
+     */
+    protected Future<Void> podDisruptionBudget() {
+        return podDisruptionBudgetOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperComponentName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletes the StrimziPodSet for the ZooKeeper cluster
+     *
+     * @return  Future which completes when the PodSet is deleted
+     */
+    protected Future<Void> podSet() {
+        return strimziPodSetOperator.deleteAsync(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperComponentName(reconciliation.name()), true);
+    }
+
+    /**
+     * Deletion of PVCs is done by honouring the deleteClaim flag.
+     * If deleteClaim is set to true (PVCs have Kafka CR as owner), this method deletes the PVCs.
+     * If deleteClaim is set to false (PVCs don't have Kafka CR as owner), this method doesn't delete the PVCs (user has to do it manually).
+     *
+      @return  Future which completes when the PVCs which should be deleted
+     */
+    protected Future<Void> deletePersistentClaims() {
+        return pvcOperator.listAsync(reconciliation.namespace(), zk.getSelectorLabels())
```

Comment:
Maybe you can use this insteaf of the `getSelectorLabels()`?

```
 Labels zkSelectorLabels = Labels.EMPTY
                    .withStrimziKind(reconciliation.kind())
                    .withStrimziCluster(reconciliation.name())
                    .withStrimziName(KafkaResources.zookeeperComponentName(reconciliation.name()));
                    ```

Reply:
Yeah it works fine as well.